qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -152,6 +169,32 @@ def preferred_init_points(self):
             self._build()
             return np.zeros(self.reps * len(self.operators), dtype=float)
 
+    def _evolve_operator(self, operator, time):
+        from qiskit.opflow import OperatorBase
+        from qiskit.extensions import HamiltonianGate
+
+        if isinstance(operator, OperatorBase):
+            if isinstance(self.evolution, EvolutionSynthesis):
```

Comment:
Would the following not make more sense?
```suggestion
            if not isinstance(self.evolution, EvolutionBase):
```